spatialprofilingtoolbox/cggnn/scripts/run_sql.py

```diff
@@ -128,6 +134,20 @@ def parse_arguments():
     return parser.parse_args()
 
 
+def retrieve_importances() -> DataFrame:
+    filename = join('out', 'importances.csv')
+    return read_csv(filename, index_col=0)
+
+
+def save_importances(_args):
+    df = retrieve_importances()
+    cohort_stratifier = _args.target_column # Is this right?
```

Comment:
The target_column isn't what the GNN is trained on, it's what's used to generate ROIs. For example, in one of the melanoma datasets, the target_column indicates whether or not a cell is cancerous, so we create ROIs focused on cancer cells only, not cells overall. The GNN is trained on patient level outcomes, which are acquired via hardcoded SQL query. We would need to change how the `cggnn` pip module acquires labels to make it a CLI argument here.

Reply:
[Specifically, see this function in cggnn.
](https://github.com/CarlinLiao/cg-gnn/blob/677c343398a2b3ea182f339390e173c5ecdf1d4d/cggnn/spt_to_df.py#L169-L185)
```
def _create_label_df(conn, specimen_study: str) -> Tuple[DataFrame, Dict[int, str]]:
    """Get slide-level results."""
    df = read_sql(f"""
        SELECT 
            slide,
            d.result
        FROM histology_assessment_process hap
            JOIN specimen_collection_process scp
                ON scp.specimen=hap.slide
            JOIN diagnosis d
                ON scp.source=d.subject
        WHERE
            scp.study='{specimen_study}';
    """, conn).set_index('slide')
    label_to_result = {i: res for i, res in enumerate(
        sort(df['result'].unique()))}
    return df.replace({res: i for i, res in label_to_result.items()}), label_to_result
```

Reply:
I see, it does not make use of the `sample_strata` that we worked out. It uses `diagnosis` directly, which probably isn't right.

In this case, it seems that there is no need for *this* new functionality to be made aware of some specific "target variable the GNN was trained on"; it should be already known as part of the study.